cfripper/rules/s3_public_access.py

```diff
@@ -45,6 +45,10 @@ def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
                 bucket_name = resource.Properties.Bucket
                 if "UNDEFINED_PARAM_" in bucket_name:
                     bucket_name = bucket_name[len("UNDEFINED_PARAM_") :]
+                if not isinstance(bucket_name, str):
+                    logger.warning(f"Not adding {type(self).__name__} failure in {logical_id} – try resolving?")
+                    continue
+
```

Comment:
I'd move this "if" above the undefined param one

Reply:
Good shout! Will do 👍